### PR TITLE
DOC: clarify OPENSSL_API_COMPAT

### LINKED ISSUE #15928: OPENSSL_API_COMPAT is no longer defined by default
In 1.1.1 we had:
```
#ifndef OPENSSL_MIN_API
# define OPENSSL_MIN_API 0
#endif

#if !defined(OPENSSL_API_COMPAT) || OPENSSL_API_COMPAT < OPENSSL_MIN_API
# undef OPENSSL_API_COMPAT
# define OPENSSL_API_COMPAT OPENSSL_MIN_API
#endif
```
In 3.0 beta 1 it says:
```
 * Applications should use -DOPENSSL_API_COMPAT=<version> to suppress the
 * declarations of functions deprecated in or before <version>.  If this is
 * undefined, the value of the macro OPENSSL_CONFIGURED_API (defined in
 * <openssl/opensslconf.h>) is the default.
```

and:
```
If not set, this macro will default to C<30000>.
```
Where that value is actually based on OPENSSL_CONFIGURED_API

But it's not defined.

### PR DESCRIPTION
**Pull request by levitte**
Fixes #15928

### CODE REVIEW

**Review by paulidale — APPROVED** [review]

File: doc/man7/openssl_user_macros.pod.in [hunk 1]
```diff
@@ -74,16 +74,18 @@ minor and patch components of the version number.  For example:
 
 =back
 
-If not set, this macro will default to
+If B<OPENSSL_API_COMPAT> is undefined, this default value is be used in its
```
Comment (t8m) [inline doc/man7/openssl_user_macros.pod.in:77]: s/is be used/is used/

**Review by t8m — CHANGES_REQUESTED** [review]

File: doc/man7/openssl_user_macros.pod.in (hunk 1 above)
Comment (levitte) [reply doc/man7/openssl_user_macros.pod.in:77]: Fixed

**Review by t8m — APPROVED** [review]

**Review by paulidale — APPROVED** [review]

**Comment by kroeckx** [comment]
> Fixes #15928

This does not fix the issue that was reported. The define did
exist by default in 1.1.1, and doesn't in 3.0.

**Comment by levitte** [comment]
@kroeckx, I saw it as a documentation issue.  It's pretty much undocumented in 1.1.1, and what there is in CHANGES says that it's a macro that the application defines, which *by consequence* means that it may absolutely be undefined.  I see the default definition in 1.1.1 as accidental, not something to be trusted... an implementation detail, if you will.

The modified documentation clarifies this.

**Comment by levitte** [comment]
Another detail is the different sort of values that `OPENSSL_API_COMPAT` may have in 3.0.  That makes it pretty bad to do easy conditionals with.

**Comment by levitte** [comment]
Merged

025312431db8c5cf0c16e1043987a7f0f9e3bf87 DOC: clarify OPENSSL_API_COMPAT

**Pull request closed without merging**